Public/Src/Engine/Scheduler/Fingerprints/PipFingerprinter.cs

```diff
@@ -50,9 +50,15 @@ public abstract class PipFingerprinter
         /// </summary>
         public delegate PipData PipDataLookup(FileArtifact artifact);
 
+        /// <summary>
+        /// Refers to a function which maps a process to its source change affected dependencies. 
+        /// </summary>
+        public delegate IReadOnlyList<AbsolutePath> SourceChangeAffectedContentsLookup(Process process);
+
         private readonly PathTable m_pathTable;
         private readonly PipFragmentRenderer.ContentHashLookup m_contentHashLookup;
         private readonly PipDataLookup m_pipDataLookup;
+        private readonly SourceChangeAffectedContentsLookup m_sourceChangeAffectedContentsLookup;

```

Comment:
It's not just sources. It's basically for all inputs to process.

Reply:
Source change affected is a attribute defining contents.